plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -192,60 +208,111 @@ private void writeTransactionEntry(PageBuilder pageBuilder, RowType entryType, T
-        // TODO: determine stats format in checkpoint based on table configuration; (https://github.com/trinodb/trino/issues/12031)
-        // currently if addFileEntry contains JSON stats we will write JSON
-        // stats to checkpoint and if addEntryFile contains parsed stats, we
-        // will write parsed stats to the checkpoint.
-        writeJsonStats(entryBlockBuilder, entryType, addFileEntry);
-        writeParsedStats(entryBlockBuilder, entryType, addFileEntry);
-        writeStringMap(entryBlockBuilder, entryType, 7, "tags", addFileEntry.getTags());
+        int fieldId = 0;
+        writeString(entryBlockBuilder, entryType, fieldId++, "path", addFileEntry.getPath());
+        writeStringMap(entryBlockBuilder, entryType, fieldId++, "partitionValues", addFileEntry.getPartitionValues());
+        writeLong(entryBlockBuilder, entryType, fieldId++, "size", addFileEntry.getSize());
+        writeLong(entryBlockBuilder, entryType, fieldId++, "modificationTime", addFileEntry.getModificationTime());
+        writeBoolean(entryBlockBuilder, entryType, fieldId++, "dataChange", addFileEntry.isDataChange());
+        if (writeStatsAsJson) {
+            writeJsonStats(entryBlockBuilder, entryType, addFileEntry, metadataEntry, fieldId++);
+        }
+        if (writeStatsAsStruct) {
+            writeParsedStats(entryBlockBuilder, entryType, addFileEntry, fieldId++);
+        }
+        writeStringMap(entryBlockBuilder, entryType, fieldId++, "tags", addFileEntry.getTags());
         blockBuilder.closeEntry();
 
         // null for others
         appendNullOtherBlocks(pageBuilder, ADD_BLOCK_CHANNEL);
     }
 
-    private void writeJsonStats(BlockBuilder entryBlockBuilder, RowType entryType, AddFileEntry addFileEntry)
+    private void writeJsonStats(BlockBuilder entryBlockBuilder, RowType entryType, AddFileEntry addFileEntry, MetadataEntry metadataEntry, int fieldId)
     {
         String statsJson = null;
-        if (addFileEntry.getStats().isPresent() && addFileEntry.getStats().get() instanceof DeltaLakeJsonFileStatistics) {
-            statsJson = addFileEntry.getStatsString().orElse(null);
+        if (addFileEntry.getStats().isPresent()) {
+            DeltaLakeFileStatistics statistics = addFileEntry.getStats().get();
+            if (statistics instanceof DeltaLakeParquetFileStatistics) {
+                Map<String, Type> columnTypeMapping = getColumnTypeMapping(metadataEntry);
+                DeltaLakeParquetFileStatistics parquetFileStatistics = (DeltaLakeParquetFileStatistics) statistics;
+                DeltaLakeJsonFileStatistics jsonFileStatistics = new DeltaLakeJsonFileStatistics(
+                        parquetFileStatistics.getNumRecords(),
+                        toJsonValues(columnTypeMapping, parquetFileStatistics.getMinValues()),
+                        toJsonValues(columnTypeMapping, parquetFileStatistics.getMaxValues()),
```

Comment:
Can you use `Optional.map` and make `toJsonValues` take/return non-optional parameters?